src/clib/lib/enkf/ecl_config.cpp

```diff
@@ -1,483 +0,0 @@
-        ecl_config_init(ecl_config, config_content);
-
-    return ecl_config;
-}
-
-ecl_config_type *ecl_config_alloc_full(bool have_eclbase, char *data_file,
-                                       ecl_grid_type *grid,
-                                       char *refcase_default,
-                                       char *sched_prediction_file) {
-    ecl_config_type *ecl_config = ecl_config_alloc_empty();
-    ecl_config->have_eclbase = have_eclbase;
-    ecl_config->grid = grid;
-    if (data_file != NULL) {
-        ecl_config_set_data_file(ecl_config, data_file);
-    }
-
-    if (refcase_default)
-        if (!ecl_config_load_refcase(ecl_config, refcase_default))
-            fprintf(stderr, "** Warning: loading refcase:%s failed \n",
-                    refcase_default);
-
-    if (sched_prediction_file)
-        ecl_config->schedule_prediction_file =
-            util_alloc_string_copy(sched_prediction_file);
-
-    return ecl_config;
-}
-
-static void handle_has_eclbase_key(ecl_config_type *ecl_config,
-                                   const config_content_type *config) {
-    /*
-     The eclbase is not internalized here; here we only flag that the
-     ECLBASE keyword has been present in the configuration. The
-     actualt value is internalized as a job_name in the model_config.
-   */
-
-    if (config_content_has_item(config, ECLBASE_KEY)) {
-        ui_return_type *ui_return = ecl_config_validate_eclbase(
-            ecl_config, config_content_iget(config, ECLBASE_KEY, 0, 0));
-        if (ui_return_get_status(ui_return) == UI_RETURN_OK)
-            ecl_config->have_eclbase = true;
-        else
-            util_abort("%s: failed to set eclbase format. Error:%s\n", __func__,
-                       ui_return_get_last_error(ui_return));
-        ui_return_free(ui_return);
-    }
-}
-
-static void handle_has_data_file_key(ecl_config_type *ecl_config,
-                                     const config_content_type *config) {
-    const char *data_file =
-        config_content_get_value_as_abspath(config, DATA_FILE_KEY);
-    ui_return_type *ui_return =
-        ecl_config_validate_data_file(ecl_config, data_file);
-    if (ui_return_get_status(ui_return) == UI_RETURN_OK)
-        ecl_config_set_data_file(ecl_config, data_file);
-    else
-        util_abort("%s: problem setting ECLIPSE data file (%s)\n", __func__,
-                   ui_return_get_last_error(ui_return));
-    ui_return_free(ui_return);
-}
-
-static void handle_has_grid_key(ecl_config_type *ecl_config,
-                                const config_content_type *config) {
-    const char *grid_file =
-        config_content_get_value_as_abspath(config, GRID_KEY);
-
-    ui_return_type *ui_return = ecl_config_validate_grid(ecl_config, grid_file);
-    if (ui_return_get_status(ui_return) == UI_RETURN_OK)
-        ecl_config_set_grid(ecl_config, grid_file);
-    else
-        util_abort("%s: failed to set grid file:%s  Error:%s \n", __func__,
```

Comment:
Abort is bad, but we should probably have similar error messages raised. I see that you check whether the file has the correct type. It would be good to add a test that checks that good error messages are given (ie. `pytest.raises(ValueError, match="{good message}"):`)